server/users.js

```diff
@@ -538,8 +539,10 @@ class User extends Chat.MessageContext {
 		// Used in punishments
 		/** @type {string} */
 		this.trackRename = '';
+		/** @type {StatusType} */
+		this.statusType = 'online';
```

Comment:
Maybe `statusType` and `userMessage` should be merged into a singular `status = {type: '', message: ''}`?

Reply:
But why?

Reply:
They're a bit related; a status has two properties: type and message. It feels weird to have them as different variables.

Reply:
but you're putting an object inside an object with no intention to use it as anything but a tuple

Reply:
Yeah, that would decrease usability for zero advantage, as far as I can tell.

Reply:
TBH, @TheJetOU's suggestion is something I might suggest as well, and I don't think the suggestion is necessarily one to be dismissed completely out of hand. The common prefix definitely makes it seems like there is a common 'status' entity that should be treated as a unit. In fact, we basically _were_ treating status as one entity in the past when it was all encode in a string (and `!` was used as the type).

Outside of the small performance overhead*, `this.status.type` vs. `this.statusType` is mostly a matter of style. You could also have a `toString()` on the `status` object Jet proposes that would encapsulate what you're doing in `getIdentityWithStatus`. Maybe my brain is warped from 5 years of writing enterprise Java, but I don't really think this is a black and white issue. I definitely use small types like this when I see common prefixes in my own code.

*_Even the performance overhead is not always cut and dry. For a class with a lot of fields always going to be megamorphic anyway, so you actually *want*  to group the fields into small types. However, performance is not really worth considering here because this is hardly a hotspot, just hard to stop myself from always thinking about it :P_

Reply:
If they're completely different, which I didn't know. I can see why they are different variables.

In hindsight, I could have inferred that from the code, since setting your status and adding a message is two different commands.

Reply:
@scheibo it would make sense if they were `statusType` and `statusMessage`, but they're `statusType` and `userMessage` specifically so people don't mistake `userMessage` as having anything to do with statuses – `userMessage` is a textual avatar, and makes more sense being grouped with `avatar` than `statusType`.

Reply:
I don't have a problem with holding related state in a single object per se (we do it for stats all the time), but it would take more than two members and/or some level of reuse (even just functions that take the combined object) for me to consider it.

Reply:
I believe when Jet's comment was made it was still named `statusMessage` and there was no public announcement of our decision to detach the two concepts, which is why I felt the need to elaborate. And I also made the same point in my third paragraph. I think we're all on the same page here - two fields is the way to go here.

Reply:
Oh, sorry, I think I missed your third paragraph. My other comment still applies – I'd really only want to group things if there were more than two things or they were obviously linked in some way, such as functions needing to pass both of them.